pygraphviz/tests/test_readwrite.py

```diff
@@ -1,39 +1,28 @@
 import pygraphviz as pgv
-import os
-import tempfile
-import pathlib
 
 
-def test_readwrite():
+def test_readwrite(tmp_path):
     A = pgv.AGraph(name="test graph")
     A.add_path([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
-    the_file = tempfile.NamedTemporaryFile(delete=False)
-    fname = the_file.name
-    # Make sure it can be opened for writing again on Win32
-    the_file.close()
+    fname = tmp_path / "file"
```

Comment:
To get the same behavior for this test, I think we need to ensure that the file is closed after it's original creation, so maybe something like:

```suggestion
    fname = tmp_path / "rw_win32_testfile"
    with open(fname) as fh:
        assert not fh.closed
    assert fh.closed
```